include/nbl/builtin/hlsl/algorithm.hlsl

```diff
@@ -90,14 +90,14 @@ struct lower_to_upper_comparator_transform_t
 
 
 template<class Accessor, class Comparator>
-uint lower_bound(inout Accessor accessor, const uint begin, const uint end, const typename Accessor::value_type value, const Comparator comp)
+uint32_t lower_bound(inout Accessor accessor, const uint32_t begin, const uint32_t end, const typename Accessor::value_type value, const Comparator comp)
 {
     impl::bound_t<Accessor,Comparator> implementation = impl::bound_t<Accessor,Comparator>::setup(begin,end,value,comp);
     return implementation(accessor);
 }
 
 template<class Accessor, class Comparator>
-uint upper_bound(inout Accessor accessor, const uint begin, const uint end, const typename Accessor::value_type value, const Comparator comp)
+uint32_t upper_bound(inout Accessor accessor, const uint32_t begin, const uint32_t end, const typename Accessor::value_type value, const Comparator comp)
```

Comment:
`inout` needs to get replaced by` NBL_ARG_REF(T)`

Reply:
I was gonna do that in a seperate PR.